portfolio/src/main/webapp/mapScript.js

```diff
@@ -167,9 +167,103 @@ function initMap() {
   map.mapTypes.set('styled_map', styledMapType);
   map.setMapTypeId('styled_map');
 
-  // Set markers on the map.
-  let markerJapanTemple = new google.maps.Marker({position: japanTemple, map: map});
-  let markerNewZealand = new google.maps.Marker({position: newZealandLake, map: map});
-  let markerUganda = new google.maps.Marker({position: ugandaView, map: map});
-  console.log("Initialised the map");
+  // Set markers on the map if marker flag is True.
+  if (markerFlag == true) {
+    let markerJapanTemple = new google.maps.Marker({position: japanTemple, map: map});
+    let markerNewZealand = new google.maps.Marker({position: newZealandLake, map: map});
+    let markerUganda = new google.maps.Marker({position: ugandaView, map: map});
+  }
+  console.log("Initialise new map");
+}
+
+function showEclipse() {
+  // Initialize new map without markers.
+  initMap(false);
+  console.log("Initialise new map");
+  
+  // Parse csv files for solar/lunar eclipses from 2020 to 2030.
+  let solarEclipse = parseCsv(readFile("./eclipse/solar_decade.csv"));
+  console.log("Parse solar eclipses file")
+
+  let lunarEclipse = parseCsv(readFile("./eclipse/lunar_decade.csv"));
+  console.log("Parse lunar eclipses file")
+  
+  // Add markers to the map.
+  for (let i = 1; i < solarEclipse.length; i++) {
+    let lat = parseLat(solarEclipse[i][9]);
+    let lng = parseLng(solarEclipse[i][10]);
```

Comment:
What are these `9`  and `10` numbers?
In such cases it's better to have named constants.